src/command_modules/azure-cli-network/azure/cli/command_modules/network/mgmt_vnet_gateway/azuredeploy.json

```diff
@@ -0,0 +1,140 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "enableBgp": {
+      "type": "bool",
+      "defaultValue": false,
+      "metadata": {
+        "description": "Enable BGP (Border Gateway Protocol)."
+      }
+    },
+    "gatewayType": {
+      "type": "string",
+      "defaultValue": "Vpn",
+      "allowedValues": [
+        "Vpn",
+        "ExpressRoute"
+      ],
+      "metadata": {
+        "description": "Gateway type."
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for resources."
+      }
+    },
+    "privateIPAllocationMethod": {
+      "type": "string",
+      "defaultValue": "dynamic",
+      "allowedValues": [
+        "static",
+        "dynamic"
+      ],
+      "metadata": {
+        "description": "Private IP address allocation."
+      }
+    },
+    "publicIpAddress": {
+      "type": "string",
+      "defaultValue": "[concat(parameters('virtualNetworkGatewayName'), 'PublicIP')]",
+      "metadata": {
+        "description": "Name or ID of public IP address to use."
+      }
+    },
+    "publicIpAddressType": {
+      "type": "string",
+      "defaultValue": "existingId",
+      "allowedValues": [
+        "existingName",
+        "existingId"
+      ],
+      "metadata": {
+        "description": "Type of Public IP Address to associate with the load balancer."
+      }
+    },
+    "sku": {
+      "type": "string",
+      "defaultValue": "Basic",
+      "allowedValues": [
+        "Basic",
+        "Standard",
+        "HighPerformance"
+      ],
+      "metadata": {
+        "description": "VPN Gateway SKU."
+      }
+    },
+    "subnetId": {
+      "type": "string",
+      "metadata": {
+        "description": "Subnet ID for the VPN gateway. Subnet must be named GatewaySubnet."
```

Comment:
yes, this is required by the service.  We could use vnet, but the documentation I've seen refers to the subnet, so I think specifying subnet is more intuitive.  We don't want to auto-create the subnet because we won't get the params right most likely and since this is an advanced scenario, the user can create their subnet separately and reference it.

---
In reply to: [74160039](https://github.com/Azure/azure-cli/pull/668#discussion_r74160039) [](ancestors = 74160039)